lib/consoletest.pm

```diff
@@ -48,13 +49,17 @@ Method executed when run() finishes and the module has result => 'fail'
 
 =cut
 sub post_fail_hook {
-    my ($self) = shift;
+    my ($self) = @_;
     $self->SUPER::post_fail_hook;
     select_console('log-console');
     $self->remount_tmp_if_ro;
     $self->export_logs_basic;
     # Export extra log after failure for further check gdm issue 1127317, also poo#45236 used for tracking action on Openqa
     $self->export_logs_desktop;
+
+    # Destroy the public cloud instance
+    select_host_console(force => 1);
+    $self->{run_args}->{my_provider}->cleanup();
```

Comment:
nice ! ( if it really works :) ) but I think you missed some `is_publiccloud()` condition , right ?

Reply:
Thank you Anton, I put the fresh verification run to the description and I added the `is_publiccloud()` condition.